Core/Core/CourseSync/CourseSyncSelector/Model/CourseSyncSelectorInteractor.swift

```diff
@@ -125,6 +133,20 @@ final class CourseSyncSelectorInteractorLive: CourseSyncSelectorInteractor {
             .eraseToAnyPublisher()
     }
 
+    func observeCourseName() -> AnyPublisher<String, Never> {
+        guard let courseID else {
+            return Just(NSLocalizedString("All Courses", comment: "")).eraseToAnyPublisher()
+        }
+
+        return courseSyncEntries
+            .map { syncEntries in
+                syncEntries.first { $0.id == courseID }?.name
+            }
+            .replaceNil(with: "")
+            .replaceError(with: "")
+            .eraseToAnyPublisher()
+    }
+
```

Comment:
Since we are not expecting the course name to change:
- I think we could name the function to something like `getObserveCourseName`.
- `courseSyncEntries` changes over time and this function will publish a lot of new value. We could change it to `courseSyncEntries.first()`.

Reply:
Good idea, changed accordingly.